### DISCUSSION TITLE
umi3子应用切换问题

### DISCUSSION THREAD

**Discussion by rendeyuwei** [umijs/umi#12505] [Q&A]
想问一下，我现在主应用下面有多个子应用，比如有子应用A和子应用B。当在主应用里面通过菜单从子应用A（比如页面是/PageA）切换到子应用B（比如页面是/PageB），我通过输出发现：由于先对/PageB进行路由匹配，/PageB的路由不在子应用A里面，所以会匹配到子应用A的404页面，然后子应用A unmount，然后mount子应用B，然后匹配到了子应用B的页面，才正常渲染。

所以存在中间会闪过一个404页面的问题，这种有什么好的办法吗。感谢🙏

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4 。
